### ISSUE TITLE
Regression in scipy build

### ISSUE THREAD

**Issue by ryanking13** [pyodide/pyodide#5131]
## 🐛 Bug

It seems like scipy build is partially broken since Oct 19~21.

For some reason, `_unuran` module is built as a static library, breaking [scipy tests](https://app.circleci.com/pipelines/github/pyodide/pyodide/7949/workflows/28cb75b6-27b8-43e2-930d-ca6afec1ac5e/jobs/115204). This often happens when `SIDE_MODULE` flag is not correctly passed when linking the shared library, but it is mysterious why is happens.

```
[735/1329] Linking target scipy/stats/_unuran/unuran_wrapper.cpython-312-wasm32-emscripten.so
emcc: warning: linking a library with `-shared` will emit a static object file.  This is a form of emulation to support existing build systems.  If you want to build a runtime shared library use the SIDE_MODULE setting. [-Wemcc]
emcc: warning: ignoring unsupported linker flag: `--version-script=/root/repo/packages/scipy/build/scipy-1.14.1/scipy/_build_utils/link-version-pyinit.map` [-Wlinkflags]
```

Since there was no change in Pyodide side, I am guessing it has something to do with build dependencies of Scipy.

I've noticed that the latest successful scipy build used meson `1.5.2` while the one with error used `1.6.0` (released 2 days ago) so it might be related somehow.

**Comment by hoodmane** [comment]
Maybe we can try capping meson in the build deps?

**Comment by ryanking13** [comment]
> Maybe we can try capping meson in the build deps?

Yeah, we can see if that works first, and then we can start finding a root cause.

**Comment by juntyr** [comment]
I'm also struggling with this issue - where would I bound the meson version?

**Comment by ryanking13** [comment]
Reopening as the root cause is not solved yet. I'll take a look soon. Thanks for the fix @juntyr!

**Comment by stephanmg** [comment]
Related: 
```
from ._unuran.unuran_wrapper import NumericalInversePolynomial
ImportError: dynamic module does not define module export function
(PyInit_unuran_wrapper)
```

Might occur if you import a (new) package you build with Pyodide in the Python REPL, like if your package imports somewhere for instance `scipy.stats.chi2` - hope this helps some other people.

**Comment by ryanking13** [comment]
Resolved by #5253

**Comment by stephanmg** [comment]
> Resolved by #5253

Thanks @ryanking13 I also can confirm that this works for me.
